--- a/target/linux/mediatek/dts/mt7981b-h3c-magic-nx30-pro.dts
+++ b/target/linux/mediatek/dts/mt7981b-h3c-magic-nx30-pro.dts
@@ -139,32 +139,6 @@
 				label = "ubi";
 				reg = <0x0580000 0x4000000>;
 			};
-
-			/* yaffs partition */
-			partition@4580000 {
-				label = "pdt_data";
-				reg = <0x4580000 0x0600000>;
-				read-only;
-			};
-
-			/* yaffs partition */
-			partition@4b80000 {
-				label = "pdt_data_1";
-				reg = <0x4b80000 0x0600000>;
-				read-only;
-			};
-
-			partition@5180000 {
-				label = "exp";
-				reg = <0x5180000 0x0100000>;
-				read-only;
-			};
-
-			partition@5280000 {
-				label = "plugin";
-				reg = <0x5280000 0x2580000>;
-				read-only;
-			};
 		};
 	};
 };

--- a/target/linux/mediatek/image/filogic.mk
+++ b/target/linux/mediatek/image/filogic.mk
@@ -208,19 +208,9 @@ define Device/h3c_magic-nx30-pro
   BLOCKSIZE := 128k
   PAGESIZE := 2048
   KERNEL_IN_UBI := 1
-  UBOOTENV_IN_UBI := 1
   IMAGE_SIZE := 65536k
-  IMAGES := sysupgrade.itb
-  KERNEL_INITRAMFS_SUFFIX := -recovery.itb
-  KERNEL := kernel-bin | gzip
-  KERNEL_INITRAMFS := kernel-bin | lzma | \
-        fit lzma $$(KDIR)/image-$$(firstword $$(DEVICE_DTS)).dtb with-initrd | pad-to 64k
-  IMAGE/sysupgrade.itb := append-kernel | \
-        fit gzip $$(KDIR)/image-$$(firstword $$(DEVICE_DTS)).dtb external-static-with-rootfs | append-metadata
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
   DEVICE_PACKAGES := kmod-mt7981-firmware mt7981-wo-firmware
-  ARTIFACTS := preloader.bin bl31-uboot.fip
-  ARTIFACT/preloader.bin := mt7981-bl2 spim-nand-ddr3
-  ARTIFACT/bl31-uboot.fip := mt7981-bl31-uboot h3c_magic-nx30-pro
 endef
 TARGET_DEVICES += h3c_magic-nx30-pro
